lib/remap-lang/src/type_def.rs

```diff
@@ -94,26 +85,15 @@ impl BitAnd for TypeDef {
     }
 }
 
-impl BitXor for TypeDef {
-    type Output = Self;
-
-    fn bitxor(self, rhs: Self) -> Self::Output {
-        let inner_type_def = match (self.inner_type_def, rhs.inner_type_def) {
-            (None, None) => None,
-            (lhs @ Some(_), None) => lhs,
-            (None, rhs @ Some(_)) => rhs,
-            (Some(lhs), Some(rhs)) => Some((*lhs ^ *rhs).boxed()),
-        };
-
-        Self {
-            fallible: self.fallible ^ rhs.fallible,
-            kind: self.kind ^ rhs.kind,
-            inner_type_def,
+impl TypeDef {
+    /// Creates a default typedef with the given Kind.
+    pub fn new_with_kind(kind: value::Kind) -> Self {
```

Comment:
`Into` doesn't always play nicely with type inference when we are chaining methods on `TypeDef`. For example, this  doesn't type check:

```
Kind::Array.into()
            .with_inner_type(inner_type_def!([ Kind::Bytes ]))
```

from https://github.com/timberio/vector/blob/f15293a83355537384ee8931a0f3e2deaeb25356/lib/remap-functions/src/parse_aws_cloudwatch_log_subscription_message.rs#L74-L75


Can you think of a nice way to make that work without falling back to `new_with_kind`? Otherwise I think we should keep the function for times when `into` won't work..

Reply:
Right. You can always do `TypeDef::from(Kind::Array).with_inner_type(...)` in that case.

From what I've seen, the method chaining happens maybe 5% of all the `TypeDef::new_with_kind` calls, so I still think it'd be a win for readability to make the change.